packages/nestjs-authorization/src/permission-set.ts

```diff
@@ -27,14 +34,14 @@ export class PermissionSet {
 
   async checkPermission(
     permission: Permission,
-    action: string,
+    action: Action,
     ctx: ExecutionContext
   ): Promise<boolean> {
     if (permission.action !== action) {
       return false;
     }
     if (permission.condition) {
-      return await permission.condition.check(ctx);
+      return await permission.condition.check(ctx, {permission, action});
```

Comment:
Why do we really need the permission object?  Would `permission.action` be enough?

Reply:
from a library point of view it's better to pass this in. but let's discuss